crates/engine/tree/src/tree/mod.rs

```diff
@@ -1391,11 +1391,14 @@ where
                 }
                 Err(err) => {
                     debug!(target: "engine", ?err, "failed to connect buffered block to tree");
+                    if let Err(fatal) = self.on_insert_block_error(err) {
+                        warn!(target: "engine", ?fatal, "fatal error occurred while connecting buffered blocks");
```

Comment:
this impls display I believe?
```suggestion
                        warn!(target: "engine", %fatal, "fatal error occurred while connecting buffered blocks");
```